### PR TITLE
Add a Dockerfile

### PR DESCRIPTION
**Pull request by AlexAndrascu**
Here is something to get this started. 

- Built & tested on `debian:buster-slim` and `ubuntu:latest`
- Build with : `docker build --squash -t dqlite:ubuntu . > build.log && tail -f build.log`
- Test with: `docker run --name dqlite-demo -p 8001:8001 -p 9001:9001 -it dqlite:ubuntu dqlite-demo -v --api 127.0.0.1:8001 --db 127.0.0.1:9001`
- Connect with `docker exec -it dqlite-demo dqlite -s 127.0.0.1:9001  test`

Resulting images and sizes (last two are the base images, for comparison): 

```
REPOSITORY      TAG             SIZE
dqlite          ubuntu           117MB
dqlite          buster-slim     111MB

ubuntu          latest             73.9MB
debian          buster-slim     69.2MB
```

### CODE REVIEW

**Comment by freeekanayaka** [comment]
Thanks!

The build failure is unrelated, it will be fixed once #266 gets merged. Going to leave a few comments to the diff inline.

File: Dockerfile
Comment (freeekanayaka) [inline Dockerfile:19]: These steps to build SQLite can be dropped. You can just install the `libsqlite3-dev` package from Debian or Ubuntu (no particular version required) and things will work.

We merged #264 today, and that gets rid of the patched version of SQLite, plain upstream SQLite is now sufficient.

File: Dockerfile
Comment (freeekanayaka) [inline Dockerfile:26]: The steps to build `libco` can be dropped too, since dqlite does not depend on it anymore starting from #264.

**Review by freeekanayaka — CHANGES_REQUESTED** [review]

File: Dockerfile
Comment (AlexAndrascu) [reply Dockerfile:26]: It fails like so if I take libco away:

```
Step 22/30 : RUN go get -d -v ./... &&     go install -tags libsqlite3 ./cmd/dqlite-demo &&     go install -tags libsqlite3 ./cmd/dqlite
 ---> Running in d73b3f842eaa
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/peterh/liner v1.2.0
go: downloading github.com/ghodss/yaml v1.0.0
go: downloading golang.org/x/sys v0.0.0-20200602225109-6fdc65e7d980
go: downloading github.com/mattn/go-sqlite3 v2.0.3+incompatible
go: downloading github.com/Rican7/retry v0.1.0
go: downloading gopkg.in/yaml.v2 v2.3.0
go: downloading github.com/spf13/cobra v1.0.0
go: downloading github.com/spf13/pflag v1.0.3
go: downloading github.com/inconshreveable/mousetrap v1.0.0
go: downloading github.com/mattn/go-runewidth v0.0.3
# github.com/canonical/go-dqlite/internal/bindings
/usr/bin/ld: cannot find -lco
collect2: error: ld returned 1 exit status
```

File: Dockerfile
Comment (AlexAndrascu) [reply Dockerfile:19]: Fixed in https://github.com/canonical/dqlite/pull/267/commits/8256b2d567559a7cc5a84a58303ffd6c81ad4a8c

Comment (AlexAndrascu) [reply Dockerfile:19]: I spoke too fast. Tried again in https://github.com/canonical/dqlite/pull/267/commits/42f0085210c706a8f8eee2872e5d744e7e54c5e3

Looking a little better:

```
REPOSITORY        TAG                       SIZE
dqlite              ubuntu                    103MB

```

File: Dockerfile
Comment (freeekanayaka) [reply Dockerfile:26]: Are you sure there's not some caching involved? I'll try myself too.

Comment (AlexAndrascu) [reply Dockerfile:26]: :) might be.  I'm still on it, haven't dropped it yet. Planning to get a docker-compose yaml up next so we can test cluster networking. Another pair of eyes wouldn't hurt tho. Can you try building with ‘--no-cache‘?

Comment (AlexAndrascu) [reply Dockerfile:26]: I have a hunch it might have to do with line no.4 from:

https://github.com/canonical/go-dqlite/blob/master/internal/bindings/build.go

Comment (freeekanayaka) [reply Dockerfile:26]: I can reproduce the issue, looking at the cause.

Comment (freeekanayaka) [reply Dockerfile:26]: > I have a hunch it might have to do with line no.4 from:
> 
> https://github.com/canonical/go-dqlite/blob/master/internal/bindings/build.go

You're right, push a PR for that.

Comment (AlexAndrascu) [reply Dockerfile:26]: Yup. Replacing:

```

package bindings

/*
#cgo linux LDFLAGS: -lsqlite3 -lraft -lco -ldqlite
*/
import "C"
```
with 

```

package bindings

/*
#cgo linux LDFLAGS: -lsqlite3 -lraft -ldqlite
*/
import "C"
```

did it. Tested locally. Any chance you can fix it there ?

Comment (freeekanayaka) [reply Dockerfile:26]: The problem should go away once https://github.com/canonical/go-dqlite/pull/120 is merged.

Comment (freeekanayaka) [reply Dockerfile:26]: That branch was merged, so your Dockerfile should work now.

Comment (AlexAndrascu) [reply Dockerfile:26]: Cleaning up and will push in a bit

**Review by freeekanayaka — APPROVED** [review]

**Pull request merged**
